spec/routes_spec.rb

```diff
@@ -74,14 +74,18 @@ def fp_message
               allow(subject).to receive(:redirect_to).with(:controller => 'dashboard', :action => 'auth_error').and_raise(MiqException::RbacPrivilegeException)
               allow(subject).to receive(:add_flash).with("The user is not authorized for this task or item.", :error).and_raise(MiqException::RbacPrivilegeException)
 
-              expect do
-                subject.send(:check_privileges)
-                subject.send(action)
-              rescue MiqException::RbacPrivilegeException
+              begin
+                expect do
+                  subject.send(:check_privileges)
+                  subject.send(action)
+                rescue MiqException::RbacPrivilegeException
+                  raise
+                end.to raise_error(MiqException::RbacPrivilegeException)
+              rescue RSpec::Expectations::ExpectationNotMetError
+                # Append a custom error message
+                $!.message << "\n\n" + error_message unless fp
```

Comment:
I don't like that this is modifying the original message...how about:

```suggestion
                "#{$!.message}\n\n#{error_message}" unless fp
```

Reply:
I can see if I can do this:

```ruby
raise RSpec::Expectations::ExpectationNotMetError, "#{$!.message}\n\n#{error_message}"
```